scala/private/phases/phase_collect_jars.bzl

```diff
@@ -0,0 +1,71 @@
+#
+# PHASE: collect jars
+#
+# DOCUMENT THIS
+#
+load(
```

Comment:
I am not quite sure what's the concerns here. All of the functions are for customizing arguments for different rules, but eventually they call the same helper function `collect_jars_from_common_ctx`.

> I'd rather have more small files than have files that change for many different reasons.

Could you elaborate more on this?

Reply:
I mean that there are two approaches I can see to organizing these calls:
1. Group all collect_jars (as well as other phases) in the same file since they call the same underlying mechanism.
2. Separate each collect_jar phase into it's own separate file. This way if the way we collect jars for the repl changes (`phase_repl_collect_jars`) junit code is unrelated (`phase_junit_test_collect_jars`).

I know that even in option 1 the diff will only show changes in the repl method but this leads to bigger files which handle a few things.

Is my intention clearer now?

Reply:
Thanks for clarification! I have thought of the second approach. The downside is we will have several small files with the same purpose, it may make it harder to compare the difference across rules. `phase_compile.bzl` will be the worse case since there are 7 rules with 7 different implementations. I can imagine if I want to make some common changes or inspect the difference, I will need to modify/open 7 files (not ideal for development).

Making subfolders may help with navigation a bit.

> even in option 1 the diff will only show changes in the repl method

I agree with this so I prefer option 1.